virtool/users/data.py

```diff
@@ -161,39 +170,98 @@ async def create(
         :param handle: the requested handle for the user
         :param password: a password
         :param force_reset: force the user to reset password on next login
+        :param  b2c_user_attributes: Azure b2c user attributes used to describe a user
         :return: the user document
         """
-        document = await create_user(self._mongo, handle, password, force_reset)
+        async with both_transactions(self._mongo, self._pg) as (mongo, pg):
+            now = virtool.utils.timestamp()
+
+            document = await create_user(
+                self._mongo,
+                handle,
+                password,
+                force_reset,
+                b2c_user_attributes=b2c_user_attributes,
+                session=mongo,
+            )
+
+            pg.add(
+                SQLUser(
+                    legacy_id=document["_id"],
+                    handle=handle,
+                    password=document["password"] if password else None,
+                    force_reset=force_reset,
+                    last_password_change=now,
+                    b2c_display_name=b2c_user_attributes.display_name
+                    if b2c_user_attributes
+                    else "",
+                    b2c_given_name=b2c_user_attributes.given_name
+                    if b2c_user_attributes
+                    else "",
+                    b2c_family_name=b2c_user_attributes.family_name
+                    if b2c_user_attributes
+                    else "",
+                    b2c_oid=b2c_user_attributes.oid if b2c_user_attributes else "",
+                )
+            )
+
         return await self.get(document["_id"])
 
+    async def delete(self, legacy_id: str):
+        """
+        Delete the user with the matching legacy_id.
+
+         :param legacy_id: legacy id of the user to delete
+         :return:
```

Comment:
Nitpick: is the left hand justification of this docstring correct? looks off to me but could be correct